app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,23 @@
+class PasswordResetsController < ApplicationController
+  def show
+    user = User.where(token: params[:id]).first
+    if user 
+      @token = user.token
+    else
+      redirect_to expired_token_path 
+    end
+  end
+
+  def create
+    user = User.where(token: params[:token]).first
+    if user
+      user.password = params[:password]
+      user.generate_token
+      user.save
```

Comment:
This process is not always right, what happens if the password by user is invalid? Also instead of set, generate and save you could use `update` like here:

```ruby
if user && update(password: params[:password])
  flash[:success] = "Your password was successfully updated."
  redirect_to sign_in_path
elsif user
  @token = user.token
  flash[:danger] = "Invalid password."
  render :show
else
  redirect_to expired_token_path
end
```